src/plugins/intel_npu/src/backend/src/zero_variable_state.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "zero_variable_state.hpp"
+
+#include "intel_npu/config/common.hpp"
+#include "intel_npu/utils/zero/zero_api.hpp"
+#include "zero_remote_tensor.hpp"
+
+namespace {
+
+template <typename Type>
+Type extract_object(const ov::AnyMap& params, const ov::Property<Type>& p) {
+    auto itrHandle = params.find(p.name());
+    ov::Any res = nullptr;
+    if (itrHandle == params.end()) {
+        OPENVINO_THROW("No parameter ", p.name(), " found in parameters map");
+    }
+    res = itrHandle->second;
+    return res.as<Type>();
+}
+
+bool memory_was_allocated_in_the_same_l0_context(ze_context_handle_t hContext, const void* ptr) {
+    ze_memory_allocation_properties_t desc = {};
+    desc.stype = ZE_STRUCTURE_TYPE_MEMORY_ALLOCATION_PROPERTIES;
+    auto res = intel_npu::zeMemGetAllocProperties(hContext, ptr, &desc, nullptr);
+    if (res == ZE_RESULT_SUCCESS) {
+        if (desc.id) {
+            if ((desc.type & ZE_MEMORY_TYPE_HOST) || (desc.type & ZE_MEMORY_TYPE_DEVICE) ||
+                (desc.type & ZE_MEMORY_TYPE_SHARED)) {
+                return true;
+            }
+        }
+    }
+
+    return false;
+}
+
+}  // namespace
+
+namespace intel_npu {
+
+ZeroVariableState::ZeroVariableState(const std::shared_ptr<ZeroInitStructsHolder>& init_structs,
+                                     const IODescriptor& descriptor,
+                                     const ov::SoPtr<ov::ITensor>& tensor,
+                                     size_t index,
```

Comment:
```suggestion
                                     size_t tensor_index,
```
Think `index` is too generic, same about `get_index`.

Reply:
done, thx